docs/bundle_format.rst

```diff
@@ -0,0 +1,106 @@
+Bundle Format
+================
+
+The Blockstore does not parse or validate most files itself. However, some conventions must be followed by clients
+to allow interoperability between environments:
+
+* There are no constraints on how the files in a Bundle can be organized.
+* OLX files must use the **.olx** extension. Files with a different extension may be ignored or not treated as
+  OLX by plugins and clients.
+* A bundle.json file must be added by authoring clients which specifies the shareable components, bundle
+  dependencies and assets. See below for details.
```

Comment:
Could you check my logic below to see if I understand the role of the `bundle.json` file for the initial work to transfer Open edX content into Blockstore?

When transferring an Unit (vertical) from Open edX to a Blockstore Bundle:
* The Unit's OLX should be the only path listed in the `components` list.
   Other OLX and related files from the Unit's XBlocks might be transferred to the Bundle too, but they will not be listed in the `components` list.
* If any of the Unit's XBlocks reference static assets, then those static assets should also be copied over to the Bundle.
  But the static assets should not be listed in the `assets` list, because they (on their own) are not "sharable" with other Bundles.
  (These static assets live in the Bundle, and so if other Units are transferred to other Bundles, then separate copies of these asset files will be stored.)
* The `dependencies` list will be empty, because the Open edX XBlock doesn't know anything about Blockstore, and so can't have dependencies on other Bundle files yet.

When transferring a Content Library from Open edX into a Blockstore Bundle:
* Each XBlock in the Library will have an OLX path in the `components` list.
* If an XBlock references static assets, then those static assets should also be copied over to the Bundle.
  But the static assets should not be included in the `assets` list, because they (on their own) are not "sharable".
* If multiple XBlocks in a Content Library reference the same asset file, then that asset file is only stored once in the Bundle.
* The `dependencies` list will be empty.

And we might have some other way to transfer just static assets from an Open edX `course` into a Blockstore Bundle:
* Each static asset will be put in the `assets` list, so that they can be shared with other Bundles.
* The `components` and `dependencies` lists would be empty.

Reply:
Yup, that is mostly accurate! I'll see if I can clarify the language in the doc even more.
 
> But the static assets should not be listed in the assets list, because they (on their own) are not "sharable" with other Bundles.

For now the purpose of the `assets` field is to list files that should be accessible by a client with a public URL. For example, if we have an html block with an image, the `.html` is a non-olx file but it does not need to be accessible from a cdn but the image file must be so that it can be fetched by browsers.

Right now, there is no mechanism for specifying non-olx files that should be shown to other authors in Bundle previews as shareable items. Am thinking about this right now.

Reply:
@symbolist  Ah cool -- I wasn't clear on what the `public` field was intended for either.  For now, I'll use `public=True` for the `asset` files, and set `public=False` for the `.html` related files, but will leave both of these out of the sharable `assets` list in `bundle.json`.

Should `.olx` files be `public` too?

Reply:
@pomegranited Being listed in `assets` is the replacement for setting `public=True`. Though this is likely going to evolve overtime as well (for example, we do not have an equivalent for asset locking as it happens in the Files & Uploads section). For now whatever makes the API happy with the least amount of work would be sufficient.